oak-core/src/main/java/org/apache/jackrabbit/oak/security/user/UserPrincipalProvider.java

```diff
@@ -209,6 +203,21 @@ public Iterator<? extends Principal> findPrincipals(int searchType) {
     }
 
     //------------------------------------------------------------< private >---
+
+    GroupPrincipalFactory createGroupPrincipalFactory() {
```

Comment:
the return value should be annotated with '@NotNull' and the methods should be private.